sydent/sydent.py

```diff
@@ -428,7 +428,7 @@ def ip_from_request(self, request):
             "http", "obey_x_forwarded_for"
         ) and request.requestHeaders.hasHeader("X-Forwarded-For"):
             return request.requestHeaders.getRawHeaders("X-Forwarded-For")[0]
-        return request.getClientIP()
+        return request.getClientAddress().host
```

Comment:
Per the [Twisted source](https://github.com/twisted/twisted/blob/release-21.2.0-10091/src/twisted/web/http.py#L1529-L1558), should we condition this on `isinstance(client, (address.IPv4Address, address.IPv6Address))` before attempting to return its `.host` property?

Reply:
@callahad: Good point, but what should be the return value, if isinstance(client, (address.IPv4Address, address.IPv6Address)) returns `False`? An empty string?

Reply:
Realistically I think it will always be an `IPv4Address` or `IPv6Address` in our setups. Once we add mypy here though it should complain...

Reply:
@clokep @callahad you find an agreement, I'll update the PR branch accordingly. Thanks.

Reply:
Twisted returns `None` if the `isinstance` check fails in `getClientIP()`, so we might as well do the same.

Reply:
@callahad: please revisit. PR branch has been amended.